### ISSUE TITLE
`if_not_else_linter()` also lints `else if` statements

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2214]
IINM, this linter is relevant only for `if`/`else` structures, and not `if`/`else if`/`else` structures. But it still produces a lint for the latter:

``` r
# dummy code
"define_cache_path <- function(cache) {
  if (isFALSE(cache)) {
    settings$cache_directory
  } else if (!is.numeric(cache)) {
    cache
  } else {
    character()
  }
}" -> code

library(lintr)
lint(text = code, linters = if_not_else_linter())
#> <text>:4:14: warning: [if_not_else_linter] In a simple if/else statement, prefer `if (A) x else y` to the less-readable `if (!A) y else x`.
#>   } else if (!is.numeric(cache)) {
#>              ^~~~~~~~~~~~~~~~~~
```

<sup>Created on 2023-09-21 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
this is intentional -- the last two branches can (and this linter argues, should) be flipped.

we intentionally excluded the "top-level"/opening condition: **if**/elseif/else, since refactoring that can be much more involved/affect readability.

**Comment by IndrajeetPatil** [comment]
Ah, okay. I was confused because I couldn't see any tests for the latter context.

Please keep this issue open because I would like further stress test this a bit and then we can close it.

**Comment by MichaelChirico** [comment]
thanks, that's definitely an omission. definitely ran those tests when writing the linter, must not have committed them.
